GitOut/Features/Git/Stage/GitStageViewModel.cs

```diff
@@ -0,0 +1,424 @@
+            }
+            SynchronizationContext? syncObject = SynchronizationContext.Current!;
+            string filename = Path.GetFileName(selectedChange.Path);
+            GitPatch? undopatch = selectedChange.Location == StatusChangeLocation.Workspace
+                ? selectedDiff.CreateUndoPatch(viewer.Selection)
+                : null;
+            GitPatch patch = selectedDiff.CreateResetPatch(viewer.Selection);
+            await Repository.ExecuteApplyAsync(patch);
+            await GetRepositoryStatusAsync(syncObject);
+
+            if (undopatch != null)
+            {
+                snack.ShowSuccess("Changes reset in " + filename, 8000, "UNDO", async () =>
+                {
+                    await Repository.ExecuteApplyAsync(undopatch);
+                    await GetRepositoryStatusAsync(syncObject);
+                });
+            }
+        }
+
+        private async Task StageSelectionAsync(FlowDocumentScrollViewer viewer)
+        {
+            if (selectedDiff is null)
+            {
+                throw new ArgumentNullException(nameof(selectedDiff), "No diff is selected");
+            }
+            if (selectedChange == null)
+            {
+                throw new ArgumentNullException(nameof(selectedChange), "No change is selected");
+            }
+            if (selectedChange.Location != StatusChangeLocation.Workspace)
+            {
+                snack.Show("Sorry, can only stage from workspace");
+                return;
+            }
+            GitPatch patch = selectedDiff.CreateAddPatch(viewer.Selection);
+            SynchronizationContext? syncObject = SynchronizationContext.Current!;
+            int previousIndex = selectedWorkspaceIndex;
+            await Repository.ExecuteApplyAsync(patch);
+            await GetRepositoryStatusAsync();
+            if (selectedChange == null)
+            {
+                SelectedWorkspaceIndex = previousIndex;
+            }
+            else
+            {
+                int index = FindSortedIndex(workspaceFiles, item => selectedChange.Path.CompareTo(item.Path));
+                if (workspaceFiles[index].Path == selectedChange.Path)
+                {
+                    await ExecuteDiffAsync(syncObject);
+                }
+            }
+        }
+
+        private async Task CommitChanges(string message)
+        {
+            await Repository.ExecuteCommitAsync(message);
+            await GetRepositoryStatusAsync();
+            snack.ShowSuccess("Commited changes successfully");
+            CommitMessage = string.Empty;
+            SelectedChange = null;
+        }
+
+        private void ParseStatus(GitStatusResult result)
+        {
+            foreach (GitStatusChange change in result.Changes)
+            {
+                if (change.IndexStatus != GitModifiedStatusType.Unmodified)
+                {
+                    if (indexFiles.Count == 0)
```

Comment:
I don't, since it's a method call to an extension that needs to check for types before figuring out what the collection is. Count == 0 is "instant" in that regard